stdlib/InteractiveUtils/src/macros.jl

```diff
@@ -342,75 +483,46 @@ function gen_call_with_extracted_types(__module__, fcn, ex0, kws = Expr[]; is_so
                 lens = map(length, rows)
                 args = Any[Expr(:tuple, lens...); vcat(rows...)]
                 ex0.head === :typed_vcat && pushfirst!(args, ex0.args[1])
-                return Expr(:call, fcn, hf, typesof_expr(args, where_params), kws...)
+                return gen_call(fcn, Any[hf, args...], where_params, kws; use_signature_tuple)
             else
-                return Expr(:call, fcn, f, typesof_expr(ex0.args, where_params), kws...)
+                return gen_call(fcn, Any[f, ex0.args...], where_params, kws; use_signature_tuple)
             end
         elseif ex0.head === :ref
-            arr = ex0.args[1]
-            ex0.args = Any[get_typeof(a) for a in ex0.args]
-            arrex = ex0.args[1]
-            if isexpr(arr, :(::), 1) || isexpr(arr, :(::), 2) || isexpr(arr, :..., 1)
-                ex0.args[1] = Expr(:call, :error, "array expression with begin/end cannot also use ::")
-            else
-                ex0.args[1] = esc(arr)
+            return expand_ref_begin_end!(ex0, __module__) do ex
+                gen_call(fcn, Any[getindex, ex.args...], where_params, kws; use_signature_tuple)
             end
-            ex, _ = replace_ref_begin_end_!(__module__, ex0, nothing, false, 1)
-            ## since replace_ref_begin_end! mutates ex0 in place, we can mutate inplace also then return ex
-            ex0.head = :call
-            ex0.args[1] = arrex
-            ex0.args = Any[fcn, Base.getindex, rewrap_where(:(Tuple{$(ex0.args...)}), where_params), kws...]
-            return ex
         else
-            PairSymAny = Pair{Symbol, Any}
-            for (head, f) in (PairSymAny(:hcat, Base.hcat),
```

Comment:
Please reapply this change to add PairSymAny, so that this part of the code can be type-stable

Reply:
Is there any actual benefit in doing so? I would mind in most cases, but not in

Reply:
I guess not, just seemed cleaner for future linters

Reply:
I think we have quite a few places (probably a lot of expression-processing functions notably) where we iterate `::Vector{Any}` anyways, instead of iterating over indices then index into the array (which I guess might be the "best" pattern for type-stability).